cktap/proto.py

```diff
@@ -234,8 +234,21 @@ def _set_derivation(self, path: List[int], cvc):
         if not all_hardened(path):
             raise ValueError("All path components must be hardened")
 
-        _, resp = self.send_auth('derive', cvc, path=path, nonce=pick_nonce())
-
+        picked_nonce = pick_nonce()
+        _, st = self.send_auth('xpub', cvc, master=True)
+        root_xpub = st['xpub']
+        pko = PubKeyNode.parse(root_xpub)
+
+        _, resp = self.send_auth('derive', cvc, path=path, nonce=picked_nonce)
+        assert resp["master_pubkey"] == pko.key
+
+        # raises on invalid sig
+        verify_master_pubkey(resp["master_pubkey"], resp["sig"], pko.chain_code,
+                             picked_nonce, self.card_nonce)
+        # repro:
+        # $ cktap derive "m/488h/589h/1h"
+        # $ cktap derive "m"  # to get empty path
+        # both above fail sig check
```

Comment:
I don't understand repro comment here. Maybe should be on an issue, not in code?